### ISSUE TITLE
Add missing required parameters with default values to workflow job when generating job execution plan

### ISSUE THREAD

**Issue by junjun-zhang** [jtracker-io/jt-wrs#5]
In workflow definition, default value may be provided to input parameter. A submitted job may have such parameter omitted, it will then default to the value provided in workflow definition.

**Comment by junjun-zhang** [comment]
Example input JSON:
```
{
    "metadata_yaml": "https://raw.githubusercontent.com/icgc-dcc/pcawg2pilot/master/tests/preprocessing/20181007/PEME-CA/WGS/CGP_donor_1199131/PD3851a/metadata.yaml"
}
```
Workflow `input` definition:
```
  input:
    metadata_yaml:
      type: string
      is_file: true
      is_required: true
    picard_jar:
      type: string
      is_file: true
      default: "[picard.jar]https://github.com/broadinstitute/picard/releases/download/2.18.16/picard.jar"
```
The job JSON misses `picard_jar` param, will need to be added. It will end up with:
```
{
    "metadata_yaml": "https://raw.githubusercontent.com/icgc-dcc/pcawg2pilot/master/tests/preprocessing/20181007/PEME-CA/WGS/CGP_donor_1199131/PD3851a/metadata.yaml",
    "picard_jar": "[${_wf_data}/picard.jar]https://github.com/broadinstitute/picard/releases/download/2.18.16/picard.jar"
}
```

Note: `${_wf_data}` is a system variable for the path to keep workflow level data files

**Comment by junjun-zhang** [comment]
addressed by #6
